lib/rubocop/cop/rspec/pending_without_reason.rb

```diff
@@ -59,61 +59,89 @@ module RSpec
-        def_node_matcher :pending_by_example_method?, block_pattern(<<~PATTERN)
-          #Examples.pending
+        # @!method skipped_in_example?(node)
+        def_node_matcher :skipped_in_example?, <<~PATTERN
+          {
+            (send nil? ${#Examples.skipped #Examples.pending})
+            (block (send nil? ${#Examples.skipped}) ...)
+            (numblock (send nil? ${#Examples.skipped}) ...)
+          }
         PATTERN
 
-        # @!method pending_by_metadata_without_reason?(node)
-        def_node_matcher :pending_by_metadata_without_reason?, <<~PATTERN
-          (send #rspec? {#ExampleGroups.all #Examples.all} ... {<(sym :pending) ...> (hash <(pair (sym :pending) true) ...>)})
+        # @!method skipped_by_example_method?(node)
+        def_node_matcher :skipped_by_example_method?, <<~PATTERN
+          (send nil? ${#Examples.skipped #Examples.pending} ...)
         PATTERN
 
-        # @!method skipped_by_example_method?(node)
-        def_node_matcher :skipped_by_example_method?, block_pattern(<<~PATTERN)
-          #Examples.skipped
+        # @!method metadata_without_reason?(node)
+        def_node_matcher :metadata_without_reason?, <<~PATTERN
+          (send #rspec?
+            {#ExampleGroups.all #Examples.all} ...
+            {
+              <(sym ${:pending :skip}) ...>
+              (hash <(pair (sym ${:pending :skip}) true) ...>)
+            }
+          )
         PATTERN
 
         # @!method skipped_by_example_group_method?(node)
-        def_node_matcher(
-          :skipped_by_example_group_method?,
-          block_pattern(<<~PATTERN)
-            #ExampleGroups.skipped
-          PATTERN
-        )
-
-        # @!method skipped_by_metadata_without_reason?(node)
-        def_node_matcher :skipped_by_metadata_without_reason?, <<~PATTERN
-          (send #rspec? {#ExampleGroups.all #Examples.all} ... {<(sym :skip) ...> (hash <(pair (sym :skip) true) ...>)})
+        def_node_matcher :skipped_by_example_group_method?, <<~PATTERN
+          (send #rspec? ${#ExampleGroups.skipped} ...)
         PATTERN
 
-        # @!method without_reason?(node)
-        def_node_matcher :without_reason?, <<~PATTERN
-          (send nil? ${:pending :skip})
+        # @!method pending_step_without_reason?(node)
+        def_node_matcher :pending_step_without_reason?, <<~PATTERN
+          (send nil? {:skip :pending})
         PATTERN
 
         def on_send(node)
-          if pending_without_reason?(node)
-            add_offense(node, message: 'Give the reason for pending.')
-          elsif skipped_without_reason?(node)
-            add_offense(node, message: 'Give the reason for skip.')
-          elsif without_reason?(node) && example?(node.parent)
-            add_offense(node,
-                        message: "Give the reason for #{node.method_name}.")
+          on_pending_by_metadata(node)
+          return unless (parent = parent_node(node))
+
+          parent = parent.begin_type? ? parent.parent : parent
```

Comment:
This line seems to belong inside `#parent_node`.

Reply:
I updated this PR. Thank you.